IonDotnet/Tree/IValueFactory.cs

```diff
@@ -0,0 +1,196 @@
+﻿using System;
+using System.Numerics;
+using IonDotnet.Tree.Impl;
+
+namespace IonDotnet.Tree
+{
+    /// <summary>
+    /// The factory for all {@link IonValue}s.
+    /// WARNING: This interface should not be implemented or extended by
+    /// code outside of this library.
+    /// </summary>
+    public interface IValueFactory
```

Comment:
These should probably be returning Interfaces, not the internal implementations.

Reply:
Done.